googletest/include/gtest/gtest_pred_impl.h

```diff
@@ -242,9 +260,15 @@ AssertionResult AssertPred4Helper(const char* pred_text,
                                   const T2& v2,
                                   const T3& v3,
                                   const T4& v4) {
-  if (pred(v1, v2, v3, v4)) return AssertionSuccess();
-
-  return AssertionFailure()
+  AssertionResult result = [&]{
+    if (pred(v1, v2, v3, v4)) {
+      return AssertionSuccess();
+    }else {
+      return AssertionFailure();
+    }
+  }();
+
+  return std::move(result)
```

Comment:
I don't believe there is an r-value reference qualified overload for `operator<<`.

Reply:
Ok, would you suggest to apply the operators first and `return std::move(...)` afterwards?

Reply:
`std::move()` shouldn't be used here at all.